.github/workflows/testing_coverage.yml

```diff
@@ -1,7 +1,13 @@
 name: pytest and coverage report
 
-# Triggers the workflow on push and PR for all branches
-on: [push, pull_request]
+# Triggers the workflow on push for all branches and PR only for main
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - main
```

Comment:
@zain-sohail: These changes you introduced after review here prevent the coverage report again from working for pull requests. This only works if the tests explicitly run for PRs in every branch. Please revert this.

Reply:
No, this means PRs _to_ main, no? Still, somehow many workflows don't run right now, and coverage reports don't work, I don't know why.

Reply:
Yes, that's indeed what it means. It should work on all pushes to whichever branch but since we always got it running twice when someone opened a PR before finishing the changes. 
This at least limits it to happen only on main. 
Can you give me examples where this workflow isn't running?

Reply:
https://github.com/OpenCOMPES/sed/pull/303 and https://github.com/OpenCOMPES/sed/pull/301 did not have coverage support. #303 Did indeed run the pull-request tests, but not the push tests, nor the linting.
In https://github.com/OpenCOMPES/sed/pull/295 it still worked, but also did not run push tests.
Not sure what the issue is.

Reply:
The reason is that coveralls does not find the base build. The reason behind this is that the respective branches are branched off the commits that bump the version after PR merges:
![grafik](https://github.com/OpenCOMPES/sed/assets/53396064/ffff8914-932c-41bc-9153-7fac457c8f74)
On these, you disabled running the tests, thus we don't have a base build in cases there are no other commits to main in-between. 
Solution: remove the exception of pyproject.toml from the test run trigger.

Reply:
I only turned it off since it was running twice but indeed, this seems like a valid reason to let the testing workflow run even with pyproject.toml change.
I will just push a change directly to main as this is an agreed upon change.